### DISCUSSION TITLE
No auto-completion

### DISCUSSION THREAD

**Discussion by romu70** [zed-industries/zed#10401] [Feature Requests]
Hi there.
I'm running an Astro project. And I'm trying to get the Astro extension to work with Zed, without success so far.

Zed finds the files are Astro, the color syntax feature is up and running, and I have no auto completion whatever I type (frontmatter, HTML, CSS or Js).

I checked the Zed log and found no error specific to the Astro language server.

Any idea?

**Comment by radianrad** [comment]
I have the same issue.

**Comment by catilius** [comment]
I had the same issue. adding typescript and astrojs language server seems to have solved the problem
```
pnpm add typescript --save-dev
pnpm add @astrojs/language-server
```
But now , Should I add this every time I start a new project ?

**Note** - In my zed.log it said
```
[ERROR] failed to start language server "astro-language-server": Request initialize failed with message: Can't find typescript.js or tsserverlibrary.js in "node_modules/typescript/lib"
[ERROR] server stderr: Some("")
```

**Comment by felixsanz** [comment]
it's solved by just installing @astrojs/language-server, but isn't that supposed to be what the extension installs?

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
